core/trino-main/src/main/java/io/trino/operator/FlatHashStrategyCompiler.java

```diff
@@ -393,9 +396,13 @@ private static void generateHashBlocksBatched(ClassDefinition definition, List<K
                     typeMethods.put(keyField.type(), method);
                 }
             }
-            body.append(invokeStatic(method, blocks.getElement(keyField.index()), hashes, offset, length));
+            computeHashes.append(invokeStatic(method, blocks.getElement(keyField.index()), hashes, offset, length));
         }
-        body.ret();
+
+        body.append(new IfStatement("if (length != 0)")
+                .condition(equal(length, constantInt(0)))
+                .ifFalse(computeHashes))
+                .ret();
```

Comment:
I think you can move this to be right after the checkFromIndexSize with an early return like this:
```
        body.append(new IfStatement("if (length == 0) return")
                        .condition(equal(length, constantInt(0)))
                        .ifTrue(RET));
```

Reply:
Sure, done.

Reply:
Turns out that sequence causes issues with the code generator because: `"JSR/RET are not supported with computeFrames option"`, so I've reverted back to the previous code shape instead.

Reply:
Oh well.  Something is broken with the library